File: backend/src/application/dto/analytics.py
"""DTO для server-side hash идентификаторов аналитики."""

from typing import Literal

from pydantic import BaseModel, Field

from src.application.dto.base import ResponseBase

AnalyticsHashKind = Literal["account", "child", "episode"]


class AnalyticsHashRequestDto(BaseModel):
    """Запрос на получение hash идентификатора для аналитики."""

    kind: AnalyticsHashKind = Field(..., description="Тип идентификатора")
    value: str = Field(..., min_length=1, max_length=256, description="Исходный id")


class AnalyticsHashResponseDto(ResponseBase):
    """Ответ с hash значением идентификатора."""

    kind: AnalyticsHashKind
    value_hash: str

File: frontend/src/shared/api/analytics.ts
import { apiClient } from "./client";

export type AnalyticsHashKind = "account" | "child" | "episode";

interface AnalyticsHashResponse {
  kind: AnalyticsHashKind;
  value_hash: string;
}

export async function hashIdentifierForAnalytics(
  kind: AnalyticsHashKind,
  value: string
): Promise<string> {
  const { data } = await apiClient.post<AnalyticsHashResponse>("/analytics/hash-identifier", {
    kind,
    value,
  });
  return data.value_hash;
}
